### ISSUE TITLE
unused-arguments-ignore-stub-functions crashes flake8

### ISSUE THREAD

**Issue by dargueta** [nhoad/flake8-unused-arguments#9]
With flake8 3.9.2 and 4.0.1, the flake8-unused-arguments plugin crashes if `unused-arguments-ignore-stub-functions` is set to `true` in the flake8 config file:

```
lint run-test-pre: PYTHONHASHSEED='2099521446'
lint run-test: commands[0] | flake8
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Users/dargueta/.pyenv/versions/3.7.8/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/Users/dargueta/.pyenv/versions/3.7.8/lib/python3.7/multiprocessing/pool.py", line 44, in mapstar
    return list(map(*args))
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/checker.py", line 676, in _run_checks
    return checker.run_checks()
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/checker.py", line 589, in run_checks
    self.run_ast_checks()
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/checker.py", line 494, in run_ast_checks
    for (line_number, offset, text, _) in runner:
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8_unused_arguments.py", line 70, in run
    if self.ignore_stubs and is_stub_function(function):
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8_unused_arguments.py", line 181, in is_stub_function
    and statement.exc.func.id == "NotImplementedError"  # type: ignore
AttributeError: 'Attribute' object has no attribute 'id'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/dargueta/dev/aircraft/.tox/lint/bin/flake8", line 8, in <module>
    sys.exit(main())
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/main/cli.py", line 22, in main
    app.run(argv)
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/main/application.py", line 363, in run
    self._run(argv)
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/main/application.py", line 351, in _run
    self.run_checks()
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/main/application.py", line 264, in run_checks
    self.file_checker_manager.run()
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/checker.py", line 321, in run
    self.run_parallel()
  File "/Users/dargueta/dev/aircraft/.tox/lint/lib/python3.7/site-packages/flake8/checker.py", line 287, in run_parallel
    for ret in pool_map:
  File "/Users/dargueta/.pyenv/versions/3.7.8/lib/python3.7/multiprocessing/pool.py", line 354, in <genexpr>
    return (item for chunk in result for item in chunk)
  File "/Users/dargueta/.pyenv/versions/3.7.8/lib/python3.7/multiprocessing/pool.py", line 748, in next
    raise value
AttributeError: 'Attribute' object has no attribute 'id'
ERROR: InvocationError for command /Users/dargueta/dev/aircraft/.tox/lint/bin/flake8 (exited with code 1)
```

**Environment:**

* Python: 3.7.8
* Flake8: 3.9.2 and 4.0.1
* Plugin: 0.0.8

**Configuration**

```ini
[flake8]
max-line-length = 88
extend-ignore = W503,W504,E203,D107,B950,G004,SIM106,SIM117,U101
exclude =
    .git,
    .tox,
    node_modules,
    .venv,
    .venv3,
    .mypy_cache,
    .pytest_cache,
    dist,
    docs
select = A,B,C,D,E,F,G,ISC,R,SIM,T4,U
max_complexity = 13
mypy_config = tox.ini
unused-arguments-ignore-stub-functions = true
```

**Comment by nhoad** [comment]
Can you provide a snippet of code that triggers this? I've got this set to true in my config file and it's not crashing.

```
$ cat ~/.config/flake8
[flake8]
max-line-length = 88

unused-arguments-ignore-abstract-functions = true
unused-arguments-ignore-stub-functions = true
$ flake8 --version
4.0.1 (flake8-unused-arguments: 0.0.8, mccabe: 0.6.1, pycodestyle: 2.8.0, pyflakes: 2.4.0) CPython 3.9.4 on Darwin
```

**Comment by dargueta** [comment]
It's a pretty large (>16 KLOC) codebase and Flake8 crashes immediately, so I have no idea what's going on.

**Comment by nhoad** [comment]
`select -U` doesn't make a difference, but I've managed to reproduce it. The trick is to define a function with a body that's only a `raise` statement, where the call isn't a function, e.g.

```
# crashes
class Foo:
    def test(self):
        raise self.error("haha nice")

# doesn't crash
class Foo:
    def test(self):
        raise Exception("haha nice")
```

This shouldn't be too difficult to fix - just need to do some more introspection. Will have a fix out this weekend.

**Comment by nhoad** [comment]
I've released 0.0.9 that fixes this. Thanks for the report!

**Comment by dargueta** [comment]
Thanks for fixing it so fast!
